python/lsst/ap/association/transformDiaSourceCatalog.py

```diff
@@ -378,12 +378,13 @@ def unpack(self, input_flag_values, flag_name):
         output_flags : `numpy.ndarray`
             Numpy named tuple of booleans.
         """
-        bit_names_types = self.output_flag_columns[flag_name]
-        output_flags = np.zeros(len(input_flag_values), dtype=bit_names_types)
+        output_flags = np.zeros(len(input_flag_values),
+                                dtype=[(name, bool) for name in self.output_flag_columns[flag_name]])
 
-        for bit_idx, (bit_name, dtypes) in enumerate(bit_names_types):
-            masked_bits = np.bitwise_and(input_flag_values, 2**bit_idx)
-            output_flags[bit_name] = masked_bits
+        for name in self.output_flag_columns[flag_name]:
+            masked_bits = np.bitwise_and(input_flag_values,
+                                         2**self.output_flag_columns[flag_name][name])
```

Comment:
I think this could be clearer if you defined `bit_idx = self.output_flag_columns[flag_name][name]` first and then set `masked_bits` using `bit_idx`, but I don't insist on it.

Reply:
It's not `bit_idx`, though, it's the bit. It's not an index.